### ISSUE TITLE
fail the installation of nuget package on xamarin.android

### ISSUE THREAD

**Issue by cansado2930** [Microsoft/project-rome#4]
Hello,
I have downloaded sample Project of Xamarin Android and I can't compile it. I have checked nuget packages and the package isn't installed. So I have tried to install it but It launchs an error. I get the same error if I try to install the nuget package on my Xamarin.Android app. The error is this:
> No se pudo instalar el paquete 'Microsoft.ConnectedDevices.Xamarin.Droid 0.1.0'. Está intentando instalar este paquete en un proyecto que tiene 'MonoAndroid,Version=v6.0' como destino, pero el paquete no contiene referencias de ensamblado o archivos de contenido compatibles con dicho marco.
Best regards

**Comment by sphenry** [comment]
Both the nuget package and the VS project target version 6.0 of Xamarin. What version do you have installed? You can check and update from Visual Studio menu Tools->Options->Xamarin

**Comment by cansado2930** [comment]
Hi @sphenry 
In theory, Xamarin is up to day, there aren't more updates in stable channel. Versions are:
Visual Studio Enterprise 2015 14.0.25431.01 Update 3
Xamarin: 4.3.0.784 (73f58d6)
Xamarin.Android 7.1.0.41 (9578cdc)
What should I check more?
Thanks

**Comment by sphenry** [comment]
That's the latest version of Xamarin, and is the same version I am using. I am on VS 2017, though. Can you also check the Project->Properties are also set to the latest version?

**Comment by cansado2930** [comment]
Ok, In order to reduce all differences, I have installed VS 2017 community on other computer and I can install the nuget package with Android 7.1. Android 7.0/6.0 launchs error. My app uses Android 19 (4.4.2) so it launchs error too. And the case of Android 7.1, I can't compile the sample app because it fires the following error:

> 1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1063,2): error MSB4018: Error inesperado en la tarea "ResolveLibraryProjectImports".

**Comment by sphenry** [comment]
As checked in, the sample app depends on the library being built and referenced from the same directory structure. I think the error you're getting is because the library path cannot be resolved (either because it hasn't been built, or can't be found). You should be able to either:
1. Build the ConnectedDevices.Xamarin.Droid and ConnectedDevices.Xamarin.Droid.Core projects
2. Remove the reference to the above libraries from Project->References and add the reference to the nugget package

**Comment by cansado2930** [comment]
Sorry by answer you later, I haven't been able to check it between week
If I compile both library project and I link them to xamarin sample, it compiles with different versions. So thanks you. I have tried to try sample app but it doesn't work. I'm going to search what is the reason. 
If there are some problem, I will notify you.
Thanks for your time.

**Comment by cmknox** [comment]
@cansado2930, Have you been able to make any progress on this?

**Comment by cansado2930** [comment]
Yes, I have a few of progress. Sample Xamarin app doesn't work. I have discover that the problem is in the webview where user sign in with MS account. I have build the Android app in order to know if there is some problem with Xamarin app but it has the same behaviour. On sample Android app shows an error message "error ocurred: invalid_request"
And I have changed the appilicationid on both projects of a real. what do you think that I have done bad?
Thanks

**Comment by mghiasi75** [comment]
I had same issues on VS2017. 
with Android SDK v6, Visual Studio can't get the NuGet package at all.

Then I changed SDK version to v7.1, this time build failed with this error:

> The "ResolveLibraryProjectImports" task failed unexpectedly.
> System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
>    at System.IO.PathHelper.GetFullPathName()
>    at System.IO.Path.LegacyNormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
>    at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
>    at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
>    at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
>    at Xamarin.Tools.Zip.ZipEntry.DoExtract(IntPtr zipFile, String destinationPath, FileMode outputFileMode, EntryExtractEventArgs args)
>    at Xamarin.Tools.Zip.ZipEntry.Extract(String destinationDir, String destinationFileName, FileMode outputFileMode)
>    at Xamarin.Android.Tools.Files.ExtractAll(ZipArchive zip, String destination, Action`2 progressCallback)
>    at Xamarin.Android.Tasks.ResolveLibraryProjectImports.Extract(DirectoryAssemblyResolver res, ICollection`1 jars, ICollection`1 resolvedResourceDirectories, ICollection`1 resolvedAssetDirectories, ICollection`1 resolvedEnvironments)
>    at Xamarin.Android.Tasks.ResolveLibraryProjectImports.Execute()
>    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
>    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()	ConnectedDevices.Xamarin.Droid.Sample			
> 

I noticed the PathTooLongException error, so I moved the folder of project to somewhere nearer to the root of drive, and it compiled successfully this time.

**TL;DR** There's a bug with NuGet package of this project with Android SDK 6 which refuses the package to install. But Android SDK 7.1 is fine. You just need to make sure the project address is not too long.

**Comment by cansado2930** [comment]
nuget package requires as @mghiasi75 have said SDK 7.1. I could compile it to SDK 6 from Project from here. A question, have you been able to use it? I tried the sample app and it shows me an error about a Request on a real device. These days I will download the new commits in order to try it again because I would like to implement it on production.

**Comment by mghiasi75** [comment]
@cansado2930 What error did you get? I opened the sample project, installed NuGet package, and deployed it to a device running Android 5.1 and it worked fine. I was able to sign in and launch a Uri from the Android device to my PC and my Windows phone.

**Comment by cansado2930** [comment]
When I compiled it 24 days ago, it launched an error About the url of login. "error Request" when I tried to do login. I have downloaded the last commit and now I can't compile "ConnectedDevices.Xamarin.Droid" :( I have attached a screenshot about errors that it shows. 
![captura](https://cloud.githubusercontent.com/assets/13149034/24933703/f962a188-1f17-11e7-8ded-691988d35786.PNG)

**Comment by sphenry** [comment]
@cansado2930 Can you show the full output from the error log?

**Comment by cansado2930** [comment]
This is the output:

> 1>------ Operación Recompilar todo iniciada: proyecto: ConnectedDevices.Xamarin.Droid.Core, configuración: Release Any CPU ------
1>  ConnectedDevices.Xamarin.Droid.Core -> C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid.Core\bin\Release\Microsoft.ConnectedDevices.Xamarin.Droid.Core.dll
2>------ Operación Recompilar todo iniciada: proyecto: ConnectedDevices.Xamarin.Droid, configuración: Release Any CPU ------
2>BINDINGSGENERATOR : warning BG8A04: <attr path="/api/package[@name='com.microsoft.connecteddevices']/class[@name='Platform.IOAuthWebAccountProviderCallback']"/> matched no nodes.
2>BINDINGSGENERATOR : warning BG8A04: <attr path="/api/package[@name='com.microsoft.connecteddevices']/class[@name='AppServiceClientConnection']/method[@name='sendMessageAsync']"/> matched no nodes.
2>BINDINGSGENERATOR : warning BG8A04: <attr path="/api/package[@name='com.microsoft.connecteddevices']/class[@name='AppServiceClientConnection']/method[@name='openRemoteAsync']"/> matched no nodes.
2>BINDINGSGENERATOR : warning BG8A04: <attr path="/api/package[@name='com.microsoft.connecteddevices']/class[@name='AppServiceClientConnection']/method[@name='openRemoteAsync']"/> matched no nodes.
2>BINDINGSGENERATOR : warning BG8A04: <attr path="/api/package[@name='com.microsoft.connecteddevices']/class[@name='AppServiceClientConnection']/method[@name='sendMessageAsync']"/> matched no nodes.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.IAuthCodeProvider.cs(26,59,26,67): warning CS0109: El miembro 'IAuthCodeProviderInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\Additions\Platform.cs(58,33,58,61): error CS0246: El nombre del tipo o del espacio de nombres 'PlatformInitializationStatus' no se encontró (¿falta una directiva using o una referencia de ensamblado?)
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\Additions\Platform.cs(49,64,49,94): error CS0535: 'Platform.PlatformInitializer' no implementa el miembro de interfaz 'IPlatformInitializationHandler.OnDone(bool)'
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.IRemoteSystemFilter.cs(21,59,21,67): warning CS0109: El miembro 'IRemoteSystemFilterInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.IPlatformInitializationHandler.cs(21,59,21,67): warning CS0109: El miembro 'IPlatformInitializationHandlerInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.IRemoteLauncherListener.cs(21,59,21,67): warning CS0109: El miembro 'IRemoteLauncherListenerInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.Platform.cs(33,60,33,68): warning CS0109: El miembro 'Platform.IAuthCodeHandlerInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
2>C:\proyecto\project-rome-master\xamarin\src\ConnectedDevices.Xamarin.Droid\obj\Release\generated\src\Microsoft.ConnectedDevices.IRemoteSystemDiscoveryListener.cs(29,59,29,67): warning CS0109: El miembro 'IRemoteSystemDiscoveryListenerInvoker._members' no oculta un miembro accesible. La palabra clave new no es necesaria.
========== Recompilar todo: 1 correctos, 1 incorrectos, 0 omitidos ==========

**Comment by sphenry** [comment]
@cansado2930 I suspect you have not download the .aar files that need to be bound. You get them by running the download-aar.ps1 in the src directory. Also, I just pushed some updates to change the min version of android required to nuget, so that package should work better for you now

**Comment by cansado2930** [comment]
Very thanks @sphenry Yesterday I downloaded the new commit and I can compile the library correctly. I didn't compile sample app but I can compile it in a new app with the library or from nuget package. So I implemented it on my production app. Thanks!!
